python-checks/src/test/resources/checks/tfFunctionRecursivity.py

```diff
@@ -0,0 +1,75 @@
+import tensorflow as tf
+
+@tf.function
+def factorial(n): # Noncompliant {{Remove this recursive call.}}
+   #^^^^^^^^^
+    if n == 1:
+        return 1
+    else:
+        return (n * factorial(n-1))
+                   #^^^^^^^^^< {{Recursive call is here.}}
+
+@tf.function
+def multiple_recursive_calls(n): # Noncompliant {{Remove this recursive call.}}
+   #^^^^^^^^^^^^^^^^^^^^^^^^
+    if cond:
+        multiple_recursive_calls(n-1)
+       #^^^^^^^^^^^^^^^^^^^^^^^^< {{Recursive call is here.}}
+    else:
+        multiple_recursive_calls(n-1)
+       #^^^^^^^^^^^^^^^^^^^^^^^^< {{Recursive call is here.}}
+    return multiple_recursive_calls(n+1) + multiple_recursive_calls(n-1)
+          #^^^^^^^^^^^^^^^^^^^^^^^^< {{Recursive call is here.}}
+                                          #^^^^^^^^^^^^^^^^^^^^^^^^@-1< {{Recursive call is here.}}
+
+@tf.function
+def indirect_rec1(n): # Noncompliant {{Remove this recursive call.}}
+   #^^^^^^^^^^^^^
+    return indirect_rec2(n)
+          #^^^^^^^^^^^^^> {{Recursive call is here.}}
+@tf.function
+def indirect_rec2(n): # Noncompliant {{Remove this recursive call.}}
+   #^^^^^^^^^^^^^
+    return indirect_rec1(n)
```

Comment:
I think this should stay out of scope given the complexity it would add but another secondary such as `Indirect recursive call.` would be really nice here if we're going to support indirect recursion.